geode-core/src/main/java/org/apache/geode/internal/cache/wan/GatewaySenderEventImpl.java

```diff
@@ -1354,4 +1355,20 @@ protected GatewaySenderEventImpl makeCopy() {
   public void setAcked(boolean acked) {
     isAcked = acked;
   }
+
+  public enum TransactionMetadataDisposition {
+    /**
+     * Transaction metadata should be excluded from the event.
+     */
+    Exclude,
+    /**
+     * Transaction metadata should be included in the event.
+     */
+    Include,
+    /**
+     * Transaction metadata should be included in the event and this is the last event in the
+     * transaction.
+     */
+    IncludeLastEvent,
```

Comment:
This name is misleading. We actually don't have a case not to include the last event. 

When tx id != null, we should always include it. The original name is "isLastEventInTransaction" which makes sense.

Maybe we should change it to "LastEventInTX"?

Reply:
I would be more include to renaming it `IncludeAsLastEvent` implying both the inclusion of the `TransactionId` and the setting of the `isLastEvent` flag.